lumicks/pylake/tests/data/mock_confocal.py

```diff
@@ -186,31 +214,347 @@ def make_slice(data):
         )
 
 
-def generate_kymo(name, image, pixel_size_nm, start=4, dt=7, samples_per_pixel=5, line_padding=3):
-    confocal_file, metadata, stop = MockConfocalFile.from_image(
+def generate_kymo(
+    name,
+    image,
+    pixel_size_nm=10.0,
+    start=20 * 10**9,
+    dt=625 * 10**8,
+    samples_per_pixel=5,
+    line_padding=3,
+):
+    return _generate_confocal(
+        name,
+        Kymo,
         image,
+        multi_color=image.ndim > 2,
         pixel_sizes_nm=[pixel_size_nm],
         axes=[0],
-        start=np.int64(start),
-        dt=np.int64(dt),
+        start=start,
+        dt=dt,
         samples_per_pixel=samples_per_pixel,
         line_padding=line_padding,
     )
 
-    return Kymo(name, confocal_file, start, stop, metadata)
+
+def _scan_axes(image, axes):
+    """Provides a central definition of default axes argument for scans"""
+    return np.arange(min(2, image.ndim)) if axes is None else axes
 
 
 def generate_scan(
-    name, scan_data, pixel_sizes_nm, start=4, dt=7, samples_per_pixel=5, line_padding=3
+    name,
+    image,
+    pixel_sizes_nm,
+    axes=None,
+    start=20 * 10**9,
+    dt=635 * 10**8,
+    samples_per_pixel=5,
+    line_padding=3,
+    multi_color=False,
 ):
+    return _generate_confocal(
+        name,
+        Scan,
+        image,
+        multi_color=multi_color,
+        pixel_sizes_nm=pixel_sizes_nm,
+        axes=_scan_axes(image, axes),
+        start=start,
+        dt=dt,
+        samples_per_pixel=samples_per_pixel,
+        line_padding=line_padding,
+    )
+
+
+def _generate_confocal(
+    name,
+    confocal_class,
+    image,
+    multi_color,
+    pixel_sizes_nm,
+    axes,
+    start,
+    dt,
+    samples_per_pixel,
+    line_padding,
+):
+    start = np.int64(start)
+    dt = np.int64(dt)
+
     confocal_file, metadata, stop = MockConfocalFile.from_image(
-        np.swapaxes(scan_data, -1, -2),
+        image,
         pixel_sizes_nm=pixel_sizes_nm,
-        axes=np.arange(scan_data.ndim),
-        start=np.int64(start),
-        dt=np.int64(dt),
+        axes=axes,
+        start=start,
+        dt=dt,
         samples_per_pixel=samples_per_pixel,
         line_padding=line_padding,
+        multi_color=multi_color,
     )
 
-    return Scan(name, confocal_file, start, stop, metadata)
+    return confocal_class(name, confocal_file, start, stop, metadata)
+
+
+def generate_kymo_with_ref(
+    name,
+    image,
+    pixel_size_nm=10.0,
+    start=20 * 10**9,
+    dt=625 * 10**8,
+    samples_per_pixel=5,
+    line_padding=3,
+):
```

Comment:
Unfortunately, we do use `generate_kymo()` and `generate_scan()` quite a lot in different tests. Otherwise, I would have liked to default to always return a reference with the function calls `generate_kymo()` and `generate_scan()`. 

This would also have had the beneficial side effect that the code for the generation of the test kymos/scans and the generation of the reference would be bundled together. The risk of having divergent parameters in how the test data would be created and what would be stored in the reference itself would be minimized. Also, the function `_scan_axes()` would not be needed to synchronize and ensure same parameters upon creation of the test data and reference.

However, splitting the test data generation and reference creation into separate functions and providing wrapper functions that return both the data and the reference, while still leaving the data generation functions marked as public, seemed a viable solution to avoid functions whose return values depend on the optional parameter `with_ref`.

Reply:
In the long run we probably could replace all calls of `generate_kymo()` and `generate_scan()`?!